rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -228,6 +236,125 @@ def _decode_curve_deposit_events(
             previous_event = event
         return DEFAULT_DECODING_OUTPUT
 
+    def _decode_curve_trades(self, context: DecoderContext) -> DecodingOutput:
+        """Decode curve trades made via single pools or curve swap router
+        First determine:
+        - `spender_address`
+        - `receiver_address`
+        - `sold_token_address`
+        - `bought_token_address`
+        - `raw_sold_amount`
+        - `raw_bought_amount`
+
+        Then create assets if `sold_token_address` and `bought_token_address` were found.
+        Then match and label events.
+        Then reshuffle events to make sure that spend and receive are consecutive.
+        """
+
+        # These are nullable because in case a curve pool is not stored in our cache or if it
+        # is a swap in a metapool (TOKEN_EXCHANGE_UNDERLYING) we will skip token check.
+        sold_token_address: Optional[ChecksumEvmAddress] = None
+        bought_token_address: Optional[ChecksumEvmAddress] = None
+
+        swapping_contract: ChecksumEvmAddress
+        if context.tx_log.topics[0] in (TOKEN_EXCHANGE, TOKEN_EXCHANGE_UNDERLYING):
+            pool_address = context.tx_log.address
+            swapping_contract = pool_address
+            # When a single pool is used, spender and receiver is always the same
+            spender_address = receiver_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+            sold_token_id = hex_or_bytes_to_int(context.tx_log.data[:32])
+            raw_sold_amount = hex_or_bytes_to_int(context.tx_log.data[32:64])
+            bought_token_id = hex_or_bytes_to_int(context.tx_log.data[64:96])
+            raw_bought_amount = hex_or_bytes_to_int(context.tx_log.data[96:128])
+            if (
+                # Can only do this check for "normal" exchanges (not exchange_unederlying) since
+                #
+                context.tx_log.topics[0] == TOKEN_EXCHANGE and  # Perform this check only for `TOKEN_EXCHANGE`  # noqa: E501
+                pool_address in self.curve_pools and  # Make sure that the pool is cached
+                len(self.curve_pools[pool_address]) > max(sold_token_id, bought_token_id)  # Make sure that tokens of the pool are cached  # noqa: E501
```

Comment:
this is because the indices are 0 indexed and then always holds `>` instead of `>=`?

Reply:
yes